### PR TITLE
feat(table-of-content): add shadow parts

### PR DESCRIPTION
**Pull request by Valentin-Sorin-Nicolae**
[ADCMS-5179](https://jsw.ibm.com/browse/ADCMS-5179)

Description

All non-slot elements in the shadow DOM should be given a unique "part" name allowing CSS to target and override component default styles. This is for the "table-of-content" component.

Changelog

New

Adding the shadow parts for the "table-of-content" component and documentation

### CODE REVIEW

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 1]
```diff
@@ -734,14 +745,16 @@ class C4DTableOfContents extends MediaQueryMixin(
                 return html`
                   <li
                     class="${itemClasses}"
+                    part="list-item list-item--active"
```
Comment (m4olivei) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:748]: Not all of these list items will be active. You need to condition `list-item--active` on wether or not the item is actually active.

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 2]
```diff
@@ -791,20 +805,32 @@ class C4DTableOfContents extends MediaQueryMixin(
             : ``}
           <div
             class="${c4dPrefix}-ce--table-of-contents__items-container"
+            part="container"
```
Comment (m4olivei) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:808]: We already have a `part="container"` for something else (line 766).
```suggestion
            part="items-container"
```

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts [hunk 3]
```diff
@@ -791,20 +805,32 @@ class C4DTableOfContents extends MediaQueryMixin(
...
             this.layout !== TOC_TYPES.HORIZONTAL
               ? `${stickyOffset}px`
               : 0}">
-            <div class="${prefix}--tableofcontents-container">
+            <div
+              class="${prefix}--tableofcontents-container"
+              part="item-container">
               <div
                 class="${prefix}--tableofcontents"
+                part="content"
```
Comment (m4olivei) [inline packages/web-components/src/components/table-of-contents/table-of-contents.ts:818]: `part="content"` is being reused a lot here. It looks to me like each place it's being used, it's a unique element that calls for a unique part name. Let's give a unique name to each.

**Review by m4olivei — CHANGES_REQUESTED** [review]

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts (hunk 1 above)
Comment (Valentin-Sorin-Nicolae) [reply packages/web-components/src/components/table-of-contents/table-of-contents.ts]: Done, as you suggested.

File: packages/web-components/src/components/table-of-contents/table-of-contents.ts (hunk 3 above)
Comment (Valentin-Sorin-Nicolae) [reply packages/web-components/src/components/table-of-contents/table-of-contents.ts:819]: Please check again.

**Review by m4olivei — APPROVED** [review]
Looks good! Thanks for all the adjustments here.

**Pull request merged**